Library/Homebrew/version.rb

```diff
@@ -703,7 +703,25 @@ def to_i
   def to_s
     version.to_s
   end
-  alias to_str to_s
+
+  sig { returns(String) }
+  def to_str
+    raise NoMethodError, "undefined method `to_str' for #{self.class}:NULL" if null?
```

Comment:
This is kinda hacky, would probably make more sense to deprecate `Version#to_str` instead.

Reply:
Sounds tricky to deprecate given implicit conversions - can't really search to see what does that and there's likely cases in formulae. And all over place really for stuff like: `Pathname("...")/version`

But doing what you're doing here for `NULL` does seem worthwhile to stop any subtle bugs.

Reply:
For the `Pathname("...")/version` case we could actually implement `Version#to_path` instead.

Reply:
Isn't exactly much of an improvement over `to_str` though.

Reply:
> Sounds tricky to deprecate given implicit conversions

Agreed 👍🏻